src/main/java/com/github/fishio/audio/FishClip.java

```diff
@@ -0,0 +1,113 @@
+package com.github.fishio.audio;
+
+import java.io.IOException;
+
+import javax.sound.sampled.Clip;
+import javax.sound.sampled.FloatControl;
+import javax.sound.sampled.LineEvent;
+import javax.sound.sampled.LineListener;
+import javax.sound.sampled.LineEvent.Type;
+
+/**
+ * A decorator for {@link Clip}, which adds functionality to check if the
+ * clip is done playing.
+ */
+public class FishClip implements LineListener, AutoCloseable {
+	private Clip clip;
+	private volatile boolean done;
+	private boolean effect;
```

Comment:
I got confused what "effect" means until I read the JavaDoc in the constructor that is a SOUND effect. So I suggest renaming it to "soundEffect".